unc_classification_tab_data/models/mlp.py

```diff
@@ -0,0 +1,252 @@
+from typing import Tuple
+
+import numpy as np
+import torch.nn as nn
+import torch
+from torch.utils.data import Dataset, DataLoader
+from .constants import *
+
+
+class MLPModule(nn.Module):
+    """Create a new multi-layer perceptron instance with ReLU activations and no non-linearity in
+    the output layer.
+
+    Parameters
+    ----------
+    model_params : dict
+        (CONSISTS OF)
+        hidden_sizes: list
+            The sizes of the hidden layers.
+        input_size: int
+            The input size.
+        output_size: int
+            The output size.
+        dropout_rate: float
+            The dropout rate applied after each layer (except the output layer)
+        batch_norm: bool
+            Whether to apply batch normalization after each layer.
+    """
+
+    def __init__(self, model_params: dict):
+        hidden_sizes = model_params.get("hidden_sizes")
+        input_size = model_params.get("input_size")
+        output_size = model_params.get("output_size", DEFAULT_OUTPUT_SIZE)
+        dropout_rate = model_params.get("dropout_rate", DEFAULT_DROPOUT_RATE)
+        do_batch_norm = model_params.get("batch_norm", DEFAULT_BATCH_NORM)
+
+        super(MLPModule, self).__init__()
+        layers = []
+
+        if len(hidden_sizes) > 0:
+            layers.append(nn.Linear(input_size, hidden_sizes[0]))
+            if do_batch_norm:
+                layers.append(nn.BatchNorm1d(num_features=hidden_sizes[0]))
+            layers.append(nn.ReLU())
+            layers.append(nn.Dropout(dropout_rate))
+
+        for i in range(1, len(hidden_sizes)):
+            layers.append(nn.Linear(hidden_sizes[i - 1], hidden_sizes[i]))
+            if do_batch_norm:
+                layers.append(nn.BatchNorm1d(num_features=hidden_sizes[i]))
+            layers.append(nn.ReLU())
+            layers.append(nn.Dropout(dropout_rate))
+
+        layers.append(nn.Linear(hidden_sizes[-1], output_size))
+
+        self.mlp = nn.Sequential(*layers)
+
+    def forward(self, _input: torch.Tensor) -> torch.Tensor:
+        """Perform a forward pass of the MLP.
+
+        Parameters
+        ----------
+        _input: torch.Tensor
+            The input of the model.
+
+        Returns
+        -------
+        type: torch.Tensor
+            The output of the model.
+        """
+        return self.mlp(_input)
+
+
+class SimpleDataset(Dataset):
```

Comment:
Should this be in `models/mlp/.py`?

Reply:
I was not sure, MLP is the only class that uses it